File: rudder-client-unity/rudder-unity-client/sdk/rudder-unity-sdk/Event/Property/ScreenPropertyBuilder.cs
using System;
namespace com.rudderlabs.unity.library.Event.Property
{
    public class ScreenPropertyBuilder : RudderPropertyBuilder
    {
        private string name;
        public ScreenPropertyBuilder SetName(string name)
        {
            this.name = name;
            return this;
        }

        public override RudderProperty Build()
        {
            if (name == null)
            {
                throw new Exception("Key \"name\" is required in properties");
            }
            RudderProperty rudderProperty = new RudderProperty();
            rudderProperty.AddProperty("name", name);
            return rudderProperty;
        }
    }
}

File: old-references/rudder-client-android-kotlin-old/src/main/java/com/rudderlabs/android/library/models/porperties/ScreenPropertyBuilder.kt
package com.rudderlabs.android.library.models.porperties

import com.rudderlabs.android.library.exceptions.MalformedEventException

class ScreenPropertyBuilder : RudderPropertyBuilder() {
    var name: String? = null
    fun setScreenName(name: String): ScreenPropertyBuilder {
        this.name = name
        return this
    }

    override fun build(): RudderProperty? {
        if (this.name == null)
            throw MalformedEventException("Key \"name\" is required in properties")
        return RudderProperty().also {
            it.addProperty("name", name)
        }
    }
}